messaging/src/main/java/org/axonframework/eventhandling/MultiSourceTrackingToken.java

```diff
@@ -76,12 +74,11 @@ public TrackingToken lowerBound(TrackingToken other) {
 
         Map<String, TrackingToken> tokenMap = new HashMap<>();
 
-        otherMultiToken.trackingTokens
-                .forEach((tokenSourceName, otherToken) ->
-                                 tokenMap.put(tokenSourceName, trackingTokens.get(tokenSourceName) == null ?
-                                                    null :
-                                                    trackingTokens.get(tokenSourceName).lowerBound(otherToken))
-        );
+        otherMultiToken.trackingTokens.forEach((tokenSourceName, otherToken) -> {
```

Comment:
I think the same issue can happen in the `upperBound` method. In that method, when the `otherToken` is null, the same can occur. We should adjust it to return the `trackingTokens.get(tokenSourceName)` if the `otherToken` is null

Reply:
Very good point! Just updated the PR to fix this. Let me know what you think!